File: app/src/main/java/com/cvv/fanstaticapps/randomticker/TickerApplication.java
package com.cvv.fanstaticapps.randomticker;

import android.app.Application;

import toothpick.Scope;
import toothpick.Toothpick;
import toothpick.configuration.Configuration;
import toothpick.registries.FactoryRegistryLocator;
import toothpick.registries.MemberInjectorRegistryLocator;
import toothpick.smoothie.module.SmoothieApplicationModule;

/**
 * Created by Carla
 * Date: 20/09/2017
 * Project: RandomTicker
 */

public class TickerApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        Toothpick.setConfiguration(Configuration.forProduction().disableReflection());
        MemberInjectorRegistryLocator.setRootRegistry(new com.cvv.fanstaticapps.randomticker.MemberInjectorRegistry());
        FactoryRegistryLocator.setRootRegistry(new com.cvv.fanstaticapps.randomticker.FactoryRegistry());

        Scope appScope = Toothpick.openScope(this);
        appScope.installModules(new SmoothieApplicationModule(this));
    }
}

File: app/src/main/java/com/cvv/fanstaticapps/randomticker/TickerApplication.kt
package com.cvv.fanstaticapps.randomticker

import com.cvv.fanstaticapps.randomticker.data.UserPreferences
import com.cvv.fanstaticapps.randomticker.injection.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication

/**
 * Created by carvaq
 * Date: 20/09/2017
 * Project: RandomTicker
 */

val PREFS: UserPreferences by lazy {
    TickerApplication.prefs!!
}

class TickerApplication : DaggerApplication() {
    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return androidInjector
    }

    companion object {
        var prefs: UserPreferences? = null
    }

    lateinit var androidInjector: AndroidInjector<out DaggerApplication>


    override fun onCreate() {
        prefs = UserPreferences(this)
        androidInjector = DaggerAppComponent
                .builder()
                .application(this)
                .build()

        super.onCreate()
    }
}
